Finished chapters 1-9

ligianicolovici opened the pull request.

ligianicolovici committed:
h2 configuration

ligianicolovici committed:
chapter 4 finished

ligianicolovici committed:
chapter 5 finished

ligianicolovici committed:
chapter 6 finished

ligianicolovici committed:
chapter 7 finished

ligianicolovici committed:
chapter 8 finished

ligianicolovici committed:
chapter 9 finished

ligianicolovici committed:
code refactoring

ligianicolovici committed:
code refactoring v2

florin-nica commented:
- Add a ".gitignore" file to tell Git which files or folders to ignore.
- Remove the unused empty "shop" folder.
- Where is the App class with the main() method?

ligianicolovici committed:
change requests - done

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/controllers/OrderController.java:
```diff
@@ -0,0 +1,22 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/order")
```
Plural Nouns Are Preferable, Rather Than Singular (for "/order" I mean)

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/converter/CSVConversion.java:
```diff
@@ -0,0 +1,42 @@
+public class CSVConversion<T> {
+    public List<T> fromCsv(Class<T> csvClass, InputStream csvData) throws IOException {
+
+        CsvMapper mapper = new CsvMapper();
```
use the mapper as a field; move it out of the methods

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/entities/Customer.java:
```diff
@@ -0,0 +1,29 @@
+@Entity
+@Builder
+@Table(name = "customer")
+
```
sometimes there is space after annotations, sometimes not. I would go with the second choice. pay attention to code formatting

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/SecurityConfiguration.java:
```diff
@@ -0,0 +1,70 @@
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import ro.msg.learning.shop.entities.Customer;
+import ro.msg.learning.shop.repositories.CustomerRepository;
+import ro.msg.learning.shop.services.CustomerService;
```
remove unused imports

florin-nica commented on shop/shop/pom.xml:
```diff
@@ -0,0 +1,213 @@
+<!--            <groupId>org.thymeleaf.extras</groupId>-->
+<!--            <artifactId>thymeleaf-extras-springsecurity5</artifactId>-->
+<!--        </dependency>-->
+
```
remove commented code

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/configuration/StrategyConfiguration.java:
```diff
@@ -0,0 +1,45 @@
+
+    private SingleLocationStrategy singleLocationStrategy = new SingleLocationStrategy();
+    private MostAbundantStrategy mostAbundantStrategy = new MostAbundantStrategy();
+    private GreedyStrategy greedyStrategy = new GreedyStrategy();
```
use dependency injection

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/configuration/StrategyConfiguration.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Value("${strategy}")
+    private String strategy;
+
```
make it of type "StrategyEnum"

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/configuration/StrategyConfiguration.java:
```diff
@@ -0,0 +1,45 @@
+                }
+            }
+        }
+        throw new ProductsCantBeShipped("Can't get place your order. We are sorry!");
```
use a switch statement by StrategyEnum types. Also could find a better name for "location selection strategies"

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/configuration/StrategyConfiguration.java:
```diff
@@ -0,0 +1,45 @@
+            if (Objects.equals(strategy.toUpperCase(), enumItem.name())) return true;
+        }
+        return false;
+    }
```
could be removed

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/converter/HttpMessageConverterCSV.java:
```diff
@@ -0,0 +1,55 @@
+package ro.msg.learning.shop.converter;
```
all other packages are using plurals for naming. change this as well to "converters"

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/controllers/ProductController.java:
```diff
@@ -0,0 +1,47 @@
+package ro.msg.learning.shop.controllers;
```
👍

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/dtos/CategoryDTO.java:
```diff
@@ -0,0 +1,18 @@
+@Builder
+@AllArgsConstructor
+public class CategoryDTO implements Serializable {
+    private String name;
```
why serializable?

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/dtos/StockDTO.java:
```diff
@@ -0,0 +1,21 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+
```
remove the empty line

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/dtos/StockDTO.java:
```diff
@@ -0,0 +1,21 @@
+    private Integer locationId;
+    private Integer productId;
+    private Integer quantity;
+
```
pay attention to code format (this includes new lines)

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/entities/Location.java:
```diff
@@ -0,0 +1,43 @@
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class Location {
+    @Id
+    @Column(unique = true)
```
do we need this annotation here?

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/entities/Location.java:
```diff
@@ -0,0 +1,43 @@
+    private String addressCountry;
+    @EqualsAndHashCode.Include
+    private String addressCity;
+    @EqualsAndHashCode.Include
```
why including only these fields? if the excluded fields fewer then maybe exclude those ones if needed and make the code cleaner

ligianicolovici replied:
had some problems at the beginning of the training with inserting data in Orders table, thought this was going to solve the problem but I discovered later the source of the problem and forgot about "cleaning" the code. 😄

florin-nica commented on shop/shop/src/main/java/ro/msg/learning/shop/mappers/OrderDetailMapper.java:
```diff
@@ -0,0 +1,55 @@
+@Component
+@RequiredArgsConstructor
+public class OrderDetailMapper {
+    private final ProductRepository productRepository;
```
remove the repository from the mapper

ligianicolovici committed:
OPT 1 JDBC - done

ligianicolovici committed:
change requests from the last PR comments - done

ligianicolovici committed:
modified single location strategy to reduce complexity + some minor other refactorings

ligianicolovici committed:
fixed a bug in singleLocationStrategy and refactored the @EqualsAndHashCode and @ToString usage

ligianicolovici committed:
- repository eliminated from OrderDetailMapper
- single and closest location refactoring by eliminating nested forEach

ligianicolovici committed:
OPT-2 MongoDB done
+ discovered and implemented fix for a use-case uncovered in product delete action.

ligianicolovici committed:
OPT-5 MVC & Thymeleaf done

ligianicolovici committed:
OPT-7 Java Mail done

ligianicolovici committed:
OPT-3 OAuth done
